R/bettr-main.R

```diff
@@ -419,265 +681,551 @@ bettr <- function(df, idCol = "Method",
+                return(NULL)
+            }
+
             shiny::validate(
-                shiny::need(collapseddata(), "")
-            )
-            scoreDf <- .calculateScores(
-                df = collapseddata(),
-                scoreMethod = input$scoreMethod,
-                idCol = idCol, scoreCol = scoreCol,
-                weightCol = weightCol, valueCol = valueCol,
-                metricCol = metricCol
-            )
-            scoreDf <- .sortAndFilterScoreData(
-                scoreDf = scoreDf,
-                idInfo = values$idInfo,
-                idCol = idCol, scoreCol = scoreCol,
-                idTopNGrouping = input$idTopNGrouping,
-                idOrdering = input$id_ordering,
-                showOnlyTopIds = input$showOnlyTopIds,
-                nbrTopIds = input$nbrTopIds
+                    shiny::need(collapseddata(), "")
+                )
+                scoreDf <- .calculateScores(
+                    df = collapseddata(),
+                    scoreMethod = if (!is.null(input$scoreMethod)) input$scoreMethod else "weighted mean",
+                    idCol = idCol, scoreCol = scoreCol,
+                    weightCol = weightCol, valueCol = valueCol,
+                    metricCol = metricCol
+                )
+                scoreDf <- .sortAndFilterScoreData(
+                    scoreDf = scoreDf,
+                    idInfo = values$idInfo,
+                    idCol = idCol, scoreCol = scoreCol,
+                    idTopNGrouping = if (!is.null(input$idTopNGrouping)) input$idTopNGrouping else "---",
+                    idOrdering = if (!is.null(input$id_ordering)) input$id_ordering else "high-to-low",
+                showOnlyTopIds = if (!is.null(input$showOnlyTopIds)) input$showOnlyTopIds else FALSE,
+                nbrTopIds = if (!is.null(input$nbrTopIds)) input$nbrTopIds else 10
             )
             scoreDf
         })
-
+        
         ## Final filtered data ------------------------------------------------
         plotdata <- shiny::reactive({
+            if (app_state$serverMode || is.null(collapseddata()) || is.null(scoredata())) {
+                return(NULL)
+            }
+
             shiny::validate(
-                shiny::need(collapseddata(), ""),
-                shiny::need(scoredata(), "")
-            )
+                    shiny::need(collapseddata(), ""),
+                    shiny::need(scoredata(), "")
+                )
             tmp <- collapseddata() |>
                 dplyr::filter(.data[[idCol]] %in% scoredata()[[idCol]])
             tmp[[idCol]] <- factor(tmp[[idCol]],
                                    levels = scoredata()[[idCol]])
             tmp
         })
-
-        observeEvent(input$autoAdjustHmHeight,
-                     shiny::updateNumericInput(
-                         session, "hmheight",
-                         value = 200L + 35L * length(unique(plotdata()[[idCol]]))
-                     ))
-
+        
```

Comment:
Why was the auto adjustment of the heatmap height removed? Or am I missing it somewhere else?